test/e2e/apiextensions_test.go

```diff
@@ -105,12 +103,43 @@ func TestCompositionPatchAndTransform(t *testing.T) {
 	)
 }
 
+// TestCompositionRealtimeRevisionSelection tests Crossplane's Composition
+// functionality to react in realtime to changes in a Composition by selecting
+// the new CompositionRevision and reconcile the XRs.
+func TestCompositionRealtimeRevisionSelection(t *testing.T) {
+	manifests := "test/e2e/manifests/apiextensions/composition/realtime-revision-selection"
+	environment.Test(t,
+		features.New(t.Name()).
+			WithLabel(LabelArea, LabelAreaAPIExtensions).
+			WithLabel(LabelSize, LabelSizeSmall).
+			WithLabel(config.LabelTestSuite, config.TestSuiteDefault).
+			WithSetup("PrerequisitesAreCreated", funcs.AllOf(
+				funcs.ApplyResources(FieldManager, manifests, "setup/*.yaml"),
+				funcs.ResourcesCreatedWithin(30*time.Second, manifests, "setup/*.yaml"),
+				funcs.ResourcesHaveConditionWithin(1*time.Minute, manifests, "setup/definition.yaml", apiextensionsv1.WatchingComposite()),
+			)).
+			Assess("CreateClaim", funcs.AllOf(
+				funcs.ApplyResources(FieldManager, manifests, "claim.yaml"),
+				funcs.ResourcesCreatedWithin(30*time.Second, manifests, "claim.yaml"),
+			)).
+			Assess("ClaimIsReady",
+				funcs.ResourcesHaveConditionWithin(5*time.Minute, manifests, "claim.yaml", xpv1.Available()),
+			).
+			Assess("UpdateComposition", funcs.AllOf(
+				funcs.ApplyResources(FieldManager, manifests, "composition-update.yaml"),
+			)).
+			Assess("ClaimHasPatchedField",
+				funcs.ResourcesHaveFieldValueWithin(10*time.Second, manifests, "claim.yaml", "status.coolerField", "I'M COOL!"),
```

Comment:
to be on the safe side, perhaps we should increase the time out to something that is slightly lower than the current composite controller polling interval (1minute)? In that way, the tests are gonna become more robust, i.e. they are not going to fail if CI is slow for some other reason.

Reply:
10s was intentional. In practice, even under load, we will be <<1s.